dds/idl/marshal_generator.cpp

```diff
@@ -2237,29 +2292,18 @@ bool marshal_generator::gen_struct(AST_Structure* node,
         "  serialized_size_list_end_parameter_id(encoding, size, xcdr1_pl_running_total);\n";
     }
   }
-
   {
     Function insertion("operator<<", "bool");
     insertion.addArg("strm", "Serializer&");
     insertion.addArg("stru", "const " + cxx + "&");
     insertion.endArgs();
 
-    if (may_be_delimited || may_be_parameter_list) {
-      be_global->impl_ <<
-        "  const Encoding& encoding = strm.encoding();\n";
```

Comment:
`encoding` is also used for getting field sizes for mutable, not just for the struct's delimiter.